### DISCUSSION TITLE
Lesson : 9 (One Big Test)

### DISCUSSION THREAD

**Discussion by today988** [Cyfrin/foundry-full-course-cu#1581] [Q&A]
Failing tests:
Encountered 1 failing test in test/unit/RaffleTest.t.sol:RaffleTest
[FAIL. Reason: setup failed: environment variable "PRIVATE KEY" not found] setUp() (gas: 0)

**Comment by EngrPips** [comment]
Hello @today988 , the error says that your private isn't found, can you please provide the private key where it is needed and rerun your test and it should pass

**Reply by today988** [reply]
I solved this problem but 2nd problem,

When i run,

forge test --fork-url $SEPOLIA_RPC_URL

I got,

Running 1 test for test/unit/RaffleTest.t.sol:RaffleTest
[FAIL. Reason: setup failed: failed parsing $PRIVATE_KEY as type uint256: missing hex prefix ("0x") for hex string] setUp() (gas: 0)
Traces:
[19424858] RaffleTest::setUp()
├─ [19294109] → new DeployRaffle@0xFEfC6BAF87cF3684058D62Da40Ff3A795946Ab06
│ └─ ← 96181 bytes of code
├─ [62488] DeployRaffle::run()
│ ├─ [25224] → new @0x6d2eed85750d316088343D6d5e91ca59eb052768
│ │ ├─ [0] VM::envUint("PRIVATE_KEY") [staticcall]
│ │ │ └─ ← failed parsing $PRIVATE_KEY as type uint256: missing hex prefix ("0x") for hex string
│ │ └─ ← 0 bytes of code
│ └─ ← failed parsing $PRIVATE_KEY as type uint256: missing hex prefix ("0x") for hex string
└─ ← failed parsing $PRIVATE_KEY as type uint256: missing hex prefix ("0x") for hex string

Test result: FAILED. 0 passed; 1 failed; 0 skipped; finished in 2.94s

Ran 1 test suites: 0 tests passed, 1 failed, 0 skipped (1 total tests)

**Reply by EngrPips** [reply]
Okay, so now all you need to do is add `0x` to the front of your private key and it should work now.

**Reply by EngrPips** [reply]
@today988 , lets assume you have your private key as shown below 

```javascript
PRIVATE_KEY=43673839048595050508696069606
```
so now you just have to make it as below 

```javascript
PRIVATE_KEY=0x43673839048595050508696069606
```
